test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php

```diff
@@ -502,4 +504,24 @@ public function testNoStubForUnknownConstant() : void
     {
         self::assertNull($this->sourceStubber->generateConstantStub('SOME_CONSTANT'));
     }
+
+    public function testStubberWithStubsDirectory(): void
+    {
+        $parser = BetterReflectionSingleton::instance()->phpParser();
+        $stubber = PhpStormStubsSourceStubber::fromStubsDirectory($parser, __DIR__.'/../../../../vendor/jetbrains/phpstorm-stubs');
+
+        $stub = $stubber->generateFunctionStub('array_filter');
+
+        self::assertInstanceOf(StubData::class, $stub);
+    }
+
+    public function testStubberThrowsExceptionWhenInvalidStubsDirectoryIsSpecified() : void
+    {
+        $parser = BetterReflectionSingleton::instance()->phpParser();
+        $stubber = PhpStormStubsSourceStubber::fromStubsDirectory($parser, '/tmp/invalid_directory');
+
+        self::expectException(InvalidFileLocation::class);
+
+        $stubber->generateFunctionStub('array_filter');
```

Comment:
Hmm, looking at this, we should throw an exception during the `fromStubsDirectory` call, not here..